### ISSUE TITLE
cafetutorial_draw_tree.py ERROR

### ISSUE THREAD

**Issue by sajjadasaf** [hahnlab/CAFE5#51]
Hello Cafe users,
I need your help as i am new to cafe.
I have run cafe5 and completed successfully. Now i want to show gene family expansion and contraction on tree by using cafetutorial_draw_tree.py script.
example command

cafetutorial_draw_tree.py -i reports/summary_run1_node.txt -t '((((cat:68.7105,horse:68.7105):4.56678,cow:73.2773):20.7227,(((((chimp:4.44417,human:4.44417):6.68268,orang:11.1268):2.28586,gibbon:13.4127):7.21153,(macaque:4.56724,baboon:4.56724):16.057):16.0607,marmoset:36.6849):57.3151):38.738,(rat:36.3024,mouse:36.3024):96.4356)' -d '((((cat<0>,horse<2>)<1>,cow<4>)<3>,(((((chimp<6>,human<8>)<7>,orang<10>)<9>,gibbon<12>)<11>,(macaque<14>,baboon<16>)<15>)<13>,marmoset<18>)<17>)<5>,(rat<20>,mouse<22>)<21>)<19>' -o reports/summary_run1_tree_rapid.png -y Rapid`

However, i am confuse about the -t  and -d options.  For -t i have pasted the tree but for -d  i am still searching.
Please help me to solve this problem.
 I am attaching my tree file  and base clade result please check it.
Any help will be appreciated

Thank you
[Base_asr.tre.txt](https://github.com/hahnlab/CAFE5/files/5879870/Base_asr.tre.txt)
[Base_clade_results.txt](https://github.com/hahnlab/CAFE5/files/5879884/Base_clade_results.txt)

**Comment by benfulton** [comment]
That script has not been updated for CAFE5 yet.

**Comment by sethbarr** [comment]
Hi @benfulton 

Any info on when this functionality will be added to CAFE5? I would also like to visualize this 

Best,

Seth

**Comment by agesantos** [comment]
Dear Fulton, 

I am trying to use CAFE for the first time. I have managed to install CAFE5 successfully and run it until the end. 

I would like to draw the tree and I have faced the problem of the issue of this thread. I considered running CAFE4 however, I have been having some troubles with the installation and I would prefer to use CAFE5, as is the most recent version. 

Will the script update be available soon? If not, could you please let me know how can I get the number of genes under Expansion, Contraction and Rapid evolution for each node of the tree from the outputs of CAFE5?  
I have been searching online and in the google group for some solutions, but I haven’t found any clear explanation. 

For instance, I have found that in CAFE4 we provide the file summary_run1_node.txt to cafetutorial_draw_tree.py. This file would be composed of a table with the flowing headers: 
Node    Expansions        Contractions     Rapidly evolving families 

As for CAFE5 I assume that the equivalent file would be base_clade_results.txt, which however has different headers: 
#Taxon_ID         Increase             Decrease

Is the file base_clade_results.txt, is “Increase” equivalent to “Expansions” of the the summary_run1_node.txt? And “Decrease” to “Contractions”? And what about Rapidly evolving families where can I find this info? 

I assume that the script would simply plot the values of the table in the tree nodes. I am correct? Or does it perform any calculations? 
If it does not perform any calculation, given that my dataset is small I can easily plot the tree manually. However, I am not sure if what I am seeing in the base_clade_results.txt is the result I need? 

I hope you can help me, 
Congratulations on this great program, 
André

**Comment by xylebori** [comment]
The easiest way to see all of the expansions and contractions by node is top open _model_clade_result.txt.  If you want them plotted on a tree it is fairly straight forward to open any one of the trees in the model_asr.tre file in Dendroscope and change the branch labels to the counts in this file (or just copy a tree from the asr.tre into a new file and edit the newick line itself).  Eventually we (Ben) will update the C code to output a tree with this data summarized at each node.  

If you want to get the family ID's that expanded or contracted at a given node you can use simple awk and grep parsing. Note I just updated the output format to remove the "+" sign from numbers in the model_change.tab file to simplify parsing, meaning you may want to recompile to use the commands below (having the "+" may or may not interfere with parsing but previously this sign was added to zero values which doesn't make sense). 

for getting expansions for the taxon or node that appears in the 6th column of your model_change.tab file:
awk '$6 > 0 {print $1,$6}' Base_change.tab > Expansions_taxonname.txt

For contractions:
awk '$6 < 0 {print $1,$6}' Base_change.tab > Contractions_taxonname.txt

To get a list of which of these are significant you parse the tree file or branch probabilities file.  

For instance Pongo abelli is in column 6, parsing the tree file to get the significant families expanded at that node looks like:  

grep "Pabel<4>\*" Base_asr.tre |awk '{print $2}' >Sig_changes_Pabel.txt

**Comment by agesantos** [comment]
Fantastic Dr Vanderpool!
Thank you for taking the time to help!

**Comment by paul-bio** [comment]
Hi @xylebori , I have a question.

First, in my *_clade_results.txt,
my target species have 1292 increased gene families.

So I followed your command above to identify the ID of gene family as below:
`awk '$8 > 0 {print $1,$8}' Base_change.tab > Expanded.fasta`

However, when I opened the Expanded.fasta file, I had 1344 gene families which were more than that of *_clade_results.txt.

To confirm gene family expansion, should I use 1344 (manually extracted) or 1292(from*_clade_results.txt)?



Best wishes.
Paul.
